### ISSUE TITLE
Problem with fetching snapshot by commit id (from version 5.6.1) when using MongoDB

### ISSUE THREAD

**Issue by patpro1** [javers/javers#958]
Update from version 5.6.0 to 5.6.1 contains change in this line in MongoRepository.java (https://github.com/javers/javers/blob/a72718afa2e0c257ec654801a03a153ae318945b/javers-persistence-mongo/src/main/java/org/javers/repository/mongo/MongoRepository.java#L282):

```
.map(c -> c.valueAsNumber().doubleValue()).collect(Collectors.toSet()));
```

was changed to:

```
.map(CommitId::valueAsNumber).collect(Collectors.toSet()));
```

After that change when snapshot has commit id with minor version different than 0 (e.g. 372136.02) this snapshot is not fetched from repository by commit id - problem is in passing BigDecimal value to query instead of double value (with double value snapshot is returned without problems - checked by debugging on line https://github.com/javers/javers/blob/a72718afa2e0c257ec654801a03a153ae318945b/javers-persistence-mongo/src/main/java/org/javers/repository/mongo/MongoRepository.java#L256)

I've added screens from debug between two version to explain the problem:

VERSION 5.6.0

<img width="824" alt="Correct behavior of version 5 6 0" src="https://user-images.githubusercontent.com/41595927/79424007-d1967100-7fbf-11ea-86a0-90ef6d1a21e6.png">

VERSION 5.6.1

<img width="824" alt="Wrong behavior of version 5 6 1" src="https://user-images.githubusercontent.com/41595927/79424022-d529f800-7fbf-11ea-86fe-cd5ce688622f.png">

**Comment by bartoszwalacik** [comment]
@patpro1 don't paste your print-screens , provide a runnable test case as stated in **Guidelines for Bug Reporting** Guidelines for Bug Reporting

https://github.com/javers/javers/blob/master/CONTRIBUTING.md#guidelines-for-bug-reporting

**Comment by bartoszwalacik** [comment]
Reopen if you want to contribute a PR or if you come up with a failing test case

**Comment by patpro1** [comment]
@bartoszwalacik I've pushed failing test to my fork: 

https://github.com/patpro1/javers/commit/4ea9eac8e1db632bc51eb161d321b192fa8a320b

I hope it's enough?

**Comment by bartoszwalacik** [comment]
fixed in 5.10.4
